src/Tools/dotnet-monitor/Auth/ApiKeyAuthenticationHandler.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Diagnostics.Monitoring.RestServer;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Extensions.Primitives;
+using Microsoft.Net.Http.Headers;
+using System;
+using System.Buffers.Text;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Net.Http.Headers;
+using System.Security.Claims;
+using System.Security.Cryptography;
+using System.Text.Encodings.Web;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Tools.Monitor
+{
+    /// <summary>
+    /// Authenticates against the ApiKey stored on the server.
+    /// </summary>
+    internal sealed class ApiKeyAuthenticationHandler : AuthenticationHandler<ApiKeyAuthenticationHandlerOptions>
+    {
+
+        public ApiKeyAuthenticationHandler(IOptionsMonitor<ApiKeyAuthenticationHandlerOptions> options, ILoggerFactory loggerFactory, UrlEncoder encoder, ISystemClock clock)
+            : base(options, loggerFactory, encoder, clock)
+        {
+        }
+
+        protected override Task<AuthenticateResult> HandleAuthenticateAsync()
+        {
+            //We are expecting a header such as Authorization: <Schema> <key>
+            //If this is not present, we will return NoResult and move on to the next authentication handler.
+            if (!Request.Headers.TryGetValue(HeaderNames.Authorization, out StringValues values) ||
+                !values.Any())
+            {
+                return Task.FromResult(AuthenticateResult.NoResult());
+            }
+
+            if (!AuthenticationHeaderValue.TryParse(values.First(), out AuthenticationHeaderValue authHeader))
+            {
+                return Task.FromResult(AuthenticateResult.Fail("Invalid authentication header"));
```

Comment:
These messages are not being shown on the curl client.

Reply:
The messages should be logged to the logging category of the handler (e.g. Microsoft.Diagnostics.Tools.Monitor.ApiKeyAuthenticationHandler) at the Information level.

But the curl client is seeing this as a 403, correct?